Code/Source/Camera/CameraSensor.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "CameraSensor.h"
+
+#include <AzCore/Math/MatrixUtils.h>
+
+#include <Atom/RPI.Public/RPISystemInterface.h>
+#include <Atom/RPI.Public/Scene.h>
+#include <Atom/RPI.Public/RenderPipeline.h>
+#include <Atom/RPI.Public/Base.h>
+
+#include <AzFramework/Components/TransformComponent.h>
+#include <AzFramework/Scene/SceneSystemInterface.h>
+
+#include <Atom/RPI.Public/Pass/Specific/RenderToTexturePass.h>
+
+#include <PostProcess/PostProcessFeatureProcessor.h>
+
+#include <Camera/CameraCaptureScheduler.h>
+
+namespace ROS2 {
+    CameraSensorDescription::CameraSensorDescription(const AZStd::string& cameraName, float verticalFov, int width, int height)
+    : verticalFieldOfViewDeg(verticalFov)
+    , width(width)
+    , height(height)
+    , cameraName(cameraName)
+    , aspectRatio(static_cast<float>(width)/static_cast<float>(height))
+    , viewToClipMatrix(MakeViewToClipMatrix()) {
+        validateParameters();
+    }
+
+    AZ::Matrix4x4 CameraSensorDescription::MakeViewToClipMatrix() const {
+        const float nearDist = 0.1f, farDist = 100.0f;
+        AZ::Matrix4x4 localViewToClipMatrix;
+        AZ::MakePerspectiveFovMatrixRH(localViewToClipMatrix, AZ::DegToRad(verticalFieldOfViewDeg), aspectRatio, nearDist, farDist, true);
+        return localViewToClipMatrix;
+    }
+
+    void CameraSensorDescription::validateParameters() const {
+        AZ_Assert(verticalFieldOfViewDeg > 0.0f && verticalFieldOfViewDeg < 180.0f, "Vertical fov should be in range 0.0 < FoV < 180.0 degrees");
+        AZ_Assert(!cameraName.empty(), "Camera name cannot be empty");
+    }
+
+    void CameraSensor::InitializePipeline(const CameraSensorDescription& cameraSensorDescription) {
+        AZ_TracePrintf("CameraSensor", "Initializing pipeline for %s", cameraSensorDescription.cameraName.c_str());
+
+        AZ::Name viewName = AZ::Name("MainCamera");
+        m_view = AZ::RPI::View::CreateView(viewName, AZ::RPI::View::UsageCamera);
+        m_view->SetViewToClipMatrix(cameraSensorDescription.viewToClipMatrix);
+        m_scene = AZ::RPI::RPISystemInterface::Get()->GetSceneByName(AZ::Name("Main"));
+
+        AZStd::string pipelineName = cameraSensorDescription.cameraName + "Pipeline";
+
+        AZ::RPI::RenderPipelineDescriptor pipelineDesc;
+        pipelineDesc.m_mainViewTagName = "MainCamera";
+        pipelineDesc.m_name = pipelineName;
+        pipelineDesc.m_rootPassTemplate = "MainPipelineRenderToTexture";
+        pipelineDesc.m_renderSettings.m_multisampleState.m_samples = 4;
```

Comment:
This is likely to affect performance, IMHO it's worth considering exposing it to users.

Reply:
Ok, this seems to be the place where this is configured. I'll leave this for now. It will take more hassle to do than I can give at the moment to do it properly. I'll fix that in the future (TODO imminent). This parameter should be exposed to the user in the same category as the near and far plane so I'll solve this later